### ISSUE TITLE
Add Button to Table

### ISSUE THREAD

**Issue by novamahend** [react-toolbox/react-toolbox#825]
Hi, btw react-toolbox is awesome.
But,  I wanna ask something. Can I add button in react-toolbox Table?, like add table in HTML table like this way:
`<table>
<tbody>
<tr>
<td><button>Click this!</button>
</tr>
</tbody>
</table>`

And one more question, how to merge two or more cell like `<td colspan="3">Column</td>` in HTML table?
Thank you.

**Comment by coljung** [comment]
@javivelasco Hey, could you let us know how it works to add React components to a table component ?

I think its possible because i see this line on TableRow.js
```javascript
if (React.isValidElement(value)) { return value; }
```

But so far i havent been able to make it work.

thx!

**Comment by rubenmoya** [comment]
I just went to the playground and loaded the table example, I changed the users array like this:

```
const users = [
  {name: 'Javi Jimenez', twitter: <Button label="@soyjavi" raised accent />, birthdate: new Date(1980, 3, 11), cats: 1},
  {name: 'Javi Velasco', twitter: <Button label="@javivelasco" raised accent />, birthdate: new Date(1987, 1, 1), dogs: 1, active: true}
];
```

And it loaded the button, so I guess that's a start :P

**Comment by novamahend** [comment]
@rubenmoya thank you so much, that's  work. And how about merging two column or more? in react-toolbox table

**Comment by coljung** [comment]
@rubenmoya that is a good example, thx.

Now, my problem is that my data is coming from an API, big JSON file, in this case i guess i would have to iterate through it, and add my React Button every time ?

**Comment by q3452** [comment]
@coljung There are proposals/PRs to add controls in the data model, which lets you define one control for the column.  See #268 or #832.  I think @javivelasco is also rethinking tables to support this functionality.
It's moving reasonably fast so wait a few days and there might be a dev branch you can pull.
If you need to fork it right now, this in TableRow.js will do the job.
**TableRow.js**
```
if (onChange) {
          return this.renderInput(key, value);
        } else if (value) {
          if (this.props.model[key].render) {
            return this.props.model[key].render(this.props.index, key, value, true);
          }
...
var inputType = _utils2.default.inputTypeForPrototype(this.props.model[key].type);
        if (this.props.model[key].render) {
          return this.props.model[key].render(index, key, value, false);
        }
```

You need the row, col and value to pass to the onChange callback from the parent container so that it can update the table source.

Your **datamodel** would need to hold something like:
```
colName: {type: String, title: 'My col', render: function(rowNum, colLabel, value, disabled) {
<Dropdown
	        auto
	        disabled={this.props.disabled}
	        label={...}
	        onChange={...}
	        source={...}
	        value={...} />
}
```

**Comment by javivelasco** [comment]
Since there are too many issues with Table customization and it started looking like a monster, I've made a new implementation that it's going to be merged to #666 if there is no negative feedback. Please check and leave your opinion!
